cogs/birthday/birthday.py

```diff
@@ -433,12 +433,49 @@ async def getSelfBirthday(self, ctx: Context):
                 if month and day:
                     birthday = date(2020, month, day)
                     birthdayStr = "{0:%B} {0:%d}".format(birthday)
-                    await ctx.author.send(
-                        f"{headerGood}: Your birthday is "
-                        f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}."
-                    )
-                    return
-
+                    try:
+                        await ctx.author.send(
+                            f"{headerGood}: Your birthday is "
+                            f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}."
+                        )
+                        return
+                    except discord.Forbidden:
+
+                        def check(msg: discord.Message):
+                            return msg.author == ctx.author and msg.channel == ctx.channel
+
+                        try:
+                            await ctx.send(
+                                "You have disabled DMs from this server. Would you still like to continue here? "
+                                "The your birthday will be sent here and deleted after a short delay. "
+                                f"Type {bold('`yes`', escape_formatting=False)} to confirm."
+                            )
+                            response = await self.bot.wait_for(
+                                "message", timeout=30.0, check=check
+                            )
+                        except asyncio.TimeoutError:
+                            await ctx.send(f"{headerBad}: No response detected. Aborting.")
+                            return
+
+                        if response.content.lower() != "yes":
+                            await ctx.send("Alright then")
+                            return
```

Comment:
I don't think this block is needed. We can just continue straight through.
```suggestion
```

Reply:
I think this is still needed because of the way the logic is structured. Without this part as long as the user types anything into the channel after reaching the warning message, it will send the birthday anyways. 

I could remove the message, but I think this logic block is necessary.